lib/prefix.c

```diff
@@ -587,6 +471,24 @@ void prefix_copy(struct prefix *dest, const struct prefix *src)
 	}
 }
 
+/* check if the two prefix_eth struct are same*/
+int prefix_eth_same(struct prefix_eth *p1, struct prefix_eth *p2)
```

Comment:
but things are handled differently for some reason.  I think this needs to be addressed by someone who understands the code better than me though.